webpack/scenes/HostCollections/Copy/CopyHostCollectionModal.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import {
+  Modal,
+  ModalVariant,
+  Form,
+  FormGroup,
+  TextInput,
+  ActionGroup,
+  Button,
+} from '@patternfly/react-core';
+import { addToast } from 'foremanReact/components/ToastsList/slice';
+import { copyHostCollection } from '../HostCollectionsActions';
+
+const CopyHostCollectionModal = ({ isOpen, onClose, hostCollection }) => {
+  const dispatch = useDispatch();
+  const [newName, setNewName] = useState(`Copy of ${hostCollection?.name || ''}`);
+  const [copying, setCopying] = useState(false);
+
+  const handleSuccess = (data) => {
+    setCopying(false);
+    dispatch(addToast({
+      type: 'success',
+      message: __('Host collection copied successfully'),
+    }));
+    onClose();
+    window.location.href = `/labs/host_collections/${data.id}`;
+  };
+
+  const handleError = (error) => {
```

Comment:
This seems like wheel-reinvention too. Tell it to look in webpack/scenes/Tasks/helpers.js and reuse stuff from there.